File: src/main/java/com/lakamsani/farefinder/Airport.java
package com.lakamsani.farefinder;

import java.util.HashMap;
import java.util.Map;

public class Airport {
    private static final Map<String, String> AIRPORTS = new HashMap<>();

    static {
        AIRPORTS.put("new york,ny", "JFK");
        AIRPORTS.put("los angeles,ca", "LAX");
        AIRPORTS.put("chicago,il", "ORD");
        AIRPORTS.put("houston,tx", "IAH");
        AIRPORTS.put("phoenix,az", "PHX");
        AIRPORTS.put("philadelphia,pa", "PHL");
        AIRPORTS.put("san antonio,tx", "SAT");
        AIRPORTS.put("san diego,ca", "SAN");
        AIRPORTS.put("dallas,tx", "DFW");
        AIRPORTS.put("san jose,ca", "SJC");
        AIRPORTS.put("austin,tx", "AUS");
        AIRPORTS.put("jacksonville,fl", "JAX");
        AIRPORTS.put("san francisco,ca", "SFO");
        AIRPORTS.put("columbus,oh", "CMH");
        AIRPORTS.put("charlotte,nc", "CLT");
        AIRPORTS.put("indianapolis,in", "IND");
        AIRPORTS.put("seattle,wa", "SEA");
        AIRPORTS.put("denver,co", "DEN");
        AIRPORTS.put("nashville,tn", "BNA");
        AIRPORTS.put("oklahoma city,ok", "OKC");
        AIRPORTS.put("el paso,tx", "ELP");
        AIRPORTS.put("washington,dc", "DCA");
        AIRPORTS.put("las vegas,nv", "LAS");
        AIRPORTS.put("louisville,ky", "SDF");
        AIRPORTS.put("baltimore,md", "BWI");
        AIRPORTS.put("milwaukee,wi", "MKE");
        AIRPORTS.put("albuquerque,nm", "ABQ");
        AIRPORTS.put("tucson,az", "TUS");
        AIRPORTS.put("fresno,ca", "FAT");
        AIRPORTS.put("sacramento,ca", "SMF");
        AIRPORTS.put("kansas city,mo", "MCI");
        AIRPORTS.put("atlanta,ga", "ATL");
        AIRPORTS.put("miami,fl", "MIA");
        AIRPORTS.put("minneapolis,mn", "MSP");
        AIRPORTS.put("portland,or", "PDX");
        AIRPORTS.put("detroit,mi", "DTW");
        AIRPORTS.put("boston,ma", "BOS");
        AIRPORTS.put("memphis,tn", "MEM");
        AIRPORTS.put("new orleans,la", "MSY");
        AIRPORTS.put("cleveland,oh", "CLE");
        AIRPORTS.put("tampa,fl", "TPA");
        AIRPORTS.put("orlando,fl", "MCO");
    }

    public static String lookupAirport(String city, String state) throws Exception {
        String key = city.strip().toLowerCase() + "," + state.strip().toLowerCase();
        String code = AIRPORTS.get(key);
        if (code == null) {
            throw new Exception("no airport found for " + city + ", " + state);
        }
        return code;
    }
}

File: kotlin/src/main/kotlin/com/lakamsani/farefinder/Airport.kt
package com.lakamsani.farefinder

/**
 * Provides IATA airport code lookup by city and state.
 */
object Airport {

    private val airports: Map<String, String> = mapOf(
        "new york,ny"       to "JFK",
        "los angeles,ca"    to "LAX",
        "chicago,il"        to "ORD",
        "houston,tx"        to "IAH",
        "phoenix,az"        to "PHX",
        "philadelphia,pa"   to "PHL",
        "san antonio,tx"    to "SAT",
        "san diego,ca"      to "SAN",
        "dallas,tx"         to "DFW",
        "san jose,ca"       to "SJC",
        "austin,tx"         to "AUS",
        "jacksonville,fl"   to "JAX",
        "san francisco,ca"  to "SFO",
        "columbus,oh"       to "CMH",
        "charlotte,nc"      to "CLT",
        "indianapolis,in"   to "IND",
        "seattle,wa"        to "SEA",
        "denver,co"         to "DEN",
        "nashville,tn"      to "BNA",
        "oklahoma city,ok"  to "OKC",
        "el paso,tx"        to "ELP",
        "washington,dc"     to "DCA",
        "las vegas,nv"      to "LAS",
        "louisville,ky"     to "SDF",
        "baltimore,md"      to "BWI",
        "milwaukee,wi"      to "MKE",
        "albuquerque,nm"    to "ABQ",
        "tucson,az"         to "TUS",
        "fresno,ca"         to "FAT",
        "sacramento,ca"     to "SMF",
        "kansas city,mo"    to "MCI",
        "atlanta,ga"        to "ATL",
        "miami,fl"          to "MIA",
        "minneapolis,mn"    to "MSP",
        "portland,or"       to "PDX",
        "detroit,mi"        to "DTW",
        "boston,ma"         to "BOS",
        "memphis,tn"        to "MEM",
        "new orleans,la"    to "MSY",
        "cleveland,oh"      to "CLE",
        "tampa,fl"          to "TPA",
        "orlando,fl"        to "MCO",
    )

    /**
     * Returns the IATA code for a given city and state.
     *
     * @param city  the city name (case-insensitive)
     * @param state the two-letter state code (case-insensitive)
     * @return the IATA airport code
     * @throws Exception if no airport is found for the city/state combination
     */
    fun lookupAirport(city: String, state: String): String {
        val key = "${city.trim().lowercase()},${state.trim().lowercase()}"
        return airports[key] ?: throw Exception("no airport found for $city, $state")
    }
}
